### PR TITLE
feat(android): add v8 heap tool to performance

### PR DESCRIPTION
**Pull request by linjinze999**
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=1880) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=1880) before we can accept your contribution.<br/><hr/>**jinzelin** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/Hippy?pullRequest=1880) it.</sub>

File: android/sdk/src/main/jni/src/bridge/performance.cc [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+using unicode_string_view = tdf::base::unicode_string_view;
+using V8VM = hippy::napi::V8VM;
+
+// [Heap] GetHeapStatistics
+jboolean GetHeapStatistics(__unused JNIEnv* j_env,
+                           __unused jobject j_object,
+                           jlong j_runtime_id,
+                           jobject j_callback) {
+    TDF_BASE_DLOG(INFO) << "GetHeapStatistics begin, j_runtime_id = " << j_runtime_id;
+    auto runtime_id = static_cast<int32_t>(j_runtime_id);
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/performance.cc:57]: 这里建议用宏check下j_runtime_id的有效性

File: android/sdk/src/main/jni/src/bridge/performance.cc [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+  j_env->DeleteLocalRef(j_cb_class);
+  // j_runtime_id invalid
+  if (!runtime) {
+      TDF_BASE_DLOG(WARNING) << "GetHeapCodeStatistics, j_runtime_id invalid";
+      j_env->CallVoidMethod(cb->GetObj(), j_cb_method, nullptr, nullptr);
+      return JNI_FALSE;
+  }
+  // prepare jni class
+  jclass j_hcs_class = j_env->FindClass("com/tencent/mtt/hippy/bridge/heap/HeapCodeStatistics");
+  jclass j_global_hcs_class = (jclass)j_env->NewGlobalRef(j_hcs_class);
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/performance.cc:133]: j_global_hcs_class为什么不用JavaRef呢

File: android/sdk/src/main/jni/src/bridge/performance.cc [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+      // v8 GetHeapCodeAndMetadataStatistics
+      auto heap_code_statistics = std::make_shared<v8::HeapCodeStatistics>();
+      v8::Isolate* isolate = std::static_pointer_cast<V8VM>(runtime->GetEngine()->GetVM())->isolate_;
+      v8::HandleScope handle_scope(isolate);
+      isolate->GetHeapCodeAndMetadataStatistics(heap_code_statistics.get());
+      // set data
+      jmethodID j_hcs_constructor = j_env->GetMethodID(j_global_hcs_class,"<init>","(JJJ)V");
+      jobject j_hcs_obj = j_env->NewObject(j_global_hcs_class,
+                                           j_hcs_constructor,
+                                           (long)heap_code_statistics->code_and_metadata_size(),
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/performance.cc:151]: c++不建议使用c风格的强转

File: android/sdk/src/main/jni/src/bridge/performance.cc [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+                                           (long)heap_code_statistics->bytecode_and_metadata_size(),
+                                           (long)heap_code_statistics->external_script_source_size());
+      j_env->CallVoidMethod(cb->GetObj(), j_cb_method, j_hcs_obj, nullptr);
+      j_env->DeleteLocalRef(j_hcs_obj);
```
Comment (churchill-zhang) [inline android/sdk/src/main/jni/src/bridge/performance.cc:155]: 用JavaRef可以避免异常漏处理，也不需要手动delete

**Comment by linjinze999** [comment]
见#2337

**Pull request closed without merging**